api/views/admin/admin.py

```diff
@@ -234,42 +241,46 @@ def read_yangcatalog_config():
         yangcatalog_config = f.read()
     response = {'info': 'Success',
                 'data': yangcatalog_config}
-    return make_response(jsonify(response), 200)
+    return response
 
 
 @app.route('/api/admin/yangcatalog-config', methods=['PUT'])
 def update_yangcatalog_config():
     current_app.logger.info('Updating yangcatalog config file')
-    body = request.json
-    input = body.get('input')
-    if input is None or input.get('data') is None:
-        return abort(400, description='payload needs to have body with "input" and "data" container')
+    body = get_input(request.json)
+    if 'data' not in body:
+        abort(400, description='"data" must be specified')
 
     with open(yc_gc.config_path, 'w') as f:
-        f.write(input['data'])
-    resp = {'api': 'error loading data',
-            'yang-search': 'error loading data',
-            'receiver': 'error loading data'}
-    yc_gc.load_config()
-    resp['api'] = 'data loaded successfully'
-    yc_gc.sender.send('reload_config')
-    resp['receiver'] = 'data loaded succesfully'
+        f.write(body['data'])
+    resp = {}
+    try:
+        yc_gc.load_config()
+        resp['api'] = 'data loaded successfully'
+    except:
+        resp['api'] = 'error loading data'
+    try:
+        yc_gc.sender.send('reload_config')
+        resp['receiver'] = 'data loaded successfully'
+    except:
+        resp['receiver'] ='error loading data'
     path = '{}://{}/yang-search/reload_config'.format(yc_gc.api_protocol, yc_gc.ip)
-    signature = create_signature(yc_gc.search_key, json.dumps(input))
+    signature = create_signature(yc_gc.search_key, json.dumps(body))
 
-    response = requests.post(path, data=json.dumps(input),
+    response = requests.post(path, data=json.dumps(body),
```

Comment:
Please remove this whole part with sending POST to yang-search/reload_config.
This functionality is no longer needed after migrating yang-search to backend.